firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift

```diff
@@ -7,62 +7,75 @@ import MozillaAppServices
 import Shared
 
 protocol BookmarksSaver {
-    /// Saves or updates a bookmark
-    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error>
+    /// Saves or updates a bookmark or folder
+    /// Returns a GUID when creating a bookmark or folder, or nil when updating them
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<GUID?, Error>
+    func createBookmark(url: String, title: String?, position: UInt32?) async
 }
 
-struct DefaultBookmarksSaver: BookmarksSaver {
+struct DefaultBookmarksSaver: BookmarksSaver, BookmarksRefactorFeatureFlagProvider {
     enum SaveError: Error {
         case bookmarkTypeDontSupportSaving
         case saveOperationFailed
     }
 
     let profile: Profile
 
-    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error> {
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<GUID?, Error> {
         return await withCheckedContinuation { continuation in
-            let operation: Success? = {
+            let operation: Deferred<Maybe<GUID?>>? = {
```

Comment:
Side note, we can avoid using Deferred in new code by replacing it with completion handlers (in the Places directly). I know this wasn't added in this PR, but just mentioning in case we need to do something similar in a next PR. I can add a follow up ticket for this work so we move away from Deferred. I don't trust a mix of `Deferred`, `withCheckedContinuation` and `async`/`await` in one function 😬

Reply:
Good call, created a [task](https://mozilla-hub.atlassian.net/browse/FXIOS-10897?jql=project%20%3D%20FXIOS%20AND%20issuetype%20%3D%20Epic%20AND%20text%20~%20%22ios%20bookmarks*%22) that I will flesh out to get work started on removing Deferred from `RushPlaces.swift`